### PR TITLE
Fix a few exploits

### PR DESCRIPTION
**Pull request by lokachop**
This should hopefully fix 2 exploits which allow users to lag/crash the server via using edited advdupe2 files or spamming framesave on EGP

### CODE REVIEW

**Comment by TomyLobo** [comment]
table.Count iterates through the entire table. I haven't looked into whether it can be replaced by # in this case, but please do so if this is a numeric-index-only table.

**Comment by Divran** [comment]
Yeah, avoid table.Count. it'd be better to keep a separate counter variable somewhere.

**Comment by lokachop** [comment]
> Yeah, avoid table.Count. it'd be better to keep a separate counter variable somewhere.

hopefully fixed now

**Review by Divran — APPROVED** [review]
Have not tested in-game but looks good

File: lua/entities/gmod_wire_egp/lib/egplib/transmitreceive.lua [hunk 1]
```diff
@@ -358,8 +358,10 @@ if (SERVER) then
 				E2.prf = E2.prf + 10
 			end
 
-			EGP:SaveFrame( E2.player, Ent, Data[1] )
-			self:AddQueue( Ent, E2.player, SaveFrame, "SaveFrame", Data[1] )
+			local status = EGP:SaveFrame( E2.player, Ent, Data[1] )
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_egp/lib/egplib/transmitreceive.lua:361]: adding an intermediate variable can help people understand the code better, but in this case, "status" isn't saying much, so idk, you could just do `if EGP:SaveFrame(...) then`.
Your call.

**Comment by TomyLobo** [comment]
looks good to me

File: lua/entities/gmod_wire_egp/lib/egplib/transmitreceive.lua (hunk 1 above)
Comment (lokachop) [reply lua/entities/gmod_wire_egp/lib/egplib/transmitreceive.lua]: local's gone now

**Pull request merged**